### PR TITLE
2404: Convert tools to ESM

### PR DESCRIPTION
**Pull request by sarahsporck**
### Short description

This PR displays in a small scope how the conversion to esm could look like. Feel free to ask questions and make remarks if you are unhappy.

### Proposed changes

- added linting rules relevant for esm
- added file extensions to import
- set type to 'module'
- update node-fetch to show that other esm packages work fine now :)

### Resolved issues

Partially addresses: #2404 

---

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -164,12 +164,23 @@ module.exports = {
       },
     },
     {
-      files: ['**/tools/**', '**/translations/**', '**/e2e-tests/**'],
+      files: ['**/tools/**', '!tools/**', '**/translations/**', '**/e2e-tests/**'],
```
[the remark this answers is not in the document]
Comment (sarahsporck) [reply .eslintrc.js:167]: All other tool directories relied on this rule, too. That means: build-configs/tools, translations/tools, web/tools, .... But I could also list those separately, if you think that's more appropriate

[the remark this answers is not in the document]
Comment (steffenkleinle) [reply .eslintrc.js:167]: Ahh no, nvm, missed that those are different patterns.

**Comment by LeandraH** [comment]
Do you have extensions that do the updated importing automatically for us? I foresee a lot of wrong imports in the future...

**Review by steffenkleinle — APPROVED** [review]

**Comment by sarahsporck** [comment]
> Do you have extensions that do the updated importing automatically for us? I foresee a lot of wrong imports in the future...

I added a eslint-plugin that checks that. Also, if you have set "type": "module" in the package.json this should done automatically by your IDE

**Review by LeandraH — APPROVED** [review]

**Pull request merged**